org.eclipse.jdt.debug.ui/ui/org/eclipse/jdt/internal/debug/ui/HotCodeReplaceErrorDialog.java

```diff
@@ -53,6 +58,7 @@ public class HotCodeReplaceErrorDialog extends ErrorDialogWithToggle {
 	public HotCodeReplaceErrorDialog(Shell parentShell, String dialogTitle, String message, IStatus status, String preferenceKey, String toggleMessage, String toggleMessage2, IPreferenceStore store, IDebugTarget target) {
 		super(parentShell, dialogTitle, message, status, preferenceKey, toggleMessage, toggleMessage2, store);
 		this.target = target;
+		DebugPlugin.getDefault().addDebugEventListener(this);
```

Comment:
This listener has to be removed after dialog is closed.

@SougandhS, @subyssurendran666 : could you please provide a fix?